screens/Login.js

```diff
@@ -0,0 +1,169 @@
+import * as React from "react";
+import {
+  Image,
+  Text,
+  View,
+  AsyncStorage,
+  StyleSheet,
+  TouchableOpacity,
+  KeyboardAvoidingView,
+  Animated,
+  Easing
+} from "react-native";
+import firebase from "firebase";
+import getGithubTokenAsync from "../config/getGithubTokenAsync";
+import GithubButton from "../components/GithubButton";
+import MyApollo from "../config/apollo";
+import CenterSpinner from "../components/CenterSpinner";
+
+const GithubStorageKey = "@Expo:GithubToken";
+
+const firebaseConfig = {
+  apiKey: "AIzaSyC73HiDOEdsS5JH8JRmX6d4wdhzJMF4Ke8",
+  authDomain: "codelab-ab6b4.firebaseapp.com",
+  projectId: "codelab-ab6b4"
+};
+
+function initializeFirebase() {
+  // Prevent reinitializing the app in snack.
+  if (!firebase.apps.length) {
+    return firebase.initializeApp(firebaseConfig);
+  }
+}
+
+async function signInAsync(token) {
+  try {
+    if (!token) {
+      const token = await getGithubTokenAsync();
+      if (token) {
+        await AsyncStorage.setItem(GithubStorageKey, token);
+        return signInAsync(token);
+      } else {
+        return;
+      }
+    }
+    const credential = firebase.auth.GithubAuthProvider.credential(token);
+    return firebase.auth().signInAndRetrieveDataWithCredential(credential);
+  } catch ({ message }) {
+    alert(message);
+  }
+}
+
+export async function signOutAsync() {
```

Comment:
Some of these functions can be put in another file, especially the `signOutAsync()` because it does even the opposite of what this file is named. You can have a `helpers` file that contains these functions.

Reply:
noted.